core/src/main/java/io/kestra/core/test/flow/Assertion.java

```diff
@@ -22,14 +28,295 @@ public class Assertion {
+                ));
+            runContext.render(this.getStartsWith()).as(String.class)
+                .ifPresent(expectedValue -> results.add(
+                    startsWith(expectedValue, actualValueQuery, actualValue, taskId, description, errorMessage)
+                ));
+            runContext.render(this.getEqualsTo()).as(Object.class)
+                .ifPresent(expectedValue -> results.add(
+                    equalsTo(expectedValue, actualValueQuery, actualValue, taskId, description, errorMessage)
+                ));
+            runContext.render(this.getContains()).as(String.class)
+                .ifPresent(expectedValue -> results.add(
+                    contains(expectedValue, actualValueQuery, actualValue, taskId, description, errorMessage)
+                ));
+            runContext.render(this.getNotEqualsTo()).as(Object.class)
+                .ifPresent(expectedValue -> results.add(
+                    notEqualsTo(expectedValue, actualValueQuery, actualValue, taskId, description, errorMessage)
+                ));
+            runContext.render(this.getGreaterThan()).as(Double.class)
+                .ifPresent(expectedValue -> results.add(
+                    greaterThan(expectedValue, actualValueQuery, actualValue, taskId, description, errorMessage)
+                ));
+            runContext.render(this.getGreaterThanOrEqualTo()).as(Double.class)
+                .ifPresent(expectedValue -> results.add(
+                    greaterThanOrEqualTo(expectedValue, actualValueQuery, actualValue, taskId, description, errorMessage))
+                );
+            runContext.render(this.getLessThan()).as(Double.class)
+                .ifPresent(expectedValue -> results.add(
+                    lessThan(expectedValue, actualValueQuery, actualValue, taskId, description, errorMessage))
+                );
+            runContext.render(this.getLessThanOrEqualTo()).as(Double.class)
+                .ifPresent(expectedValue -> results.add(
+                    lessThanOrEqualTo(expectedValue, actualValueQuery, actualValue, taskId, description, errorMessage))
+                );
+
+            var expectedInList = runContext.render(this.getIn()).asList(String.class);
+            if (!expectedInList.isEmpty()) {
+                results.add(in(expectedInList, actualValueQuery, actualValue, taskId, description, errorMessage));
+            }
+            var notExpectedInList = runContext.render(this.getNotIn()).asList(String.class);
+            if (!notExpectedInList.isEmpty()) {
+                results.add(notIn(notExpectedInList, actualValueQuery, actualValue, taskId, description, errorMessage));
+            }
+
+            if (results.isEmpty()) {
+                throw new IllegalArgumentException("no assertion to run found");
+            }
+            return results;
+        } catch (IllegalVariableEvaluationException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
And this is why an ERROR state would be great ;) 
If you cannot evaluate the assertion, you should end the suite in ERROR (which is different from a FAILED assertion).

Reply:
will do in following pr

Reply:
https://github.com/kestra-io/kestra-ee/issues/3667